opm/simulators/wells/TargetCalculator.cpp

```diff
@@ -95,29 +95,35 @@ RateType TargetCalculator::calcModeRateFromRates(const RateType* rates) const
     }
 }
 
-double TargetCalculator::groupTarget(const Group::ProductionControls ctrl) const
+double TargetCalculator::groupTarget(const std::optional<Group::ProductionControls>& ctrl, Opm::DeferredLogger& deferred_logger) const
 {
+    if (!ctrl && !use_gpmaint_) {
+        OPM_DEFLOG_THROW(std::logic_error,
+                         "Production group " + this->group_name_
+                         + "must either have a valid control or use GPMAINT",
+                         deferred_logger);
+    }
     switch (cmode_) {
     case Group::ProductionCMode::ORAT:
-        return ctrl.oil_target;
+        return ctrl->oil_target;
```

Comment:
Out of curiosity: if gpmaint is used then none of these modes in the cases apply?

Reply:
Yes that is correct.